src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -16913,5 +16982,146 @@ struct S
                 //         s.P.ToString();
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s.P").WithLocation(8, 9));
         }
+
+        [Fact]
+        public void UnassignedClassAutoProperty()
+        {
+            var source =
+@"class C
+{
+    object? P { get; }
+    C(out object o)
+    {
+        o = P;
+        P.ToString();
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (6,13): warning CS8601: Possible null reference assignment.
+                //         o = P;
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "P").WithLocation(6, 13),
+                // (7,9): warning CS8602: Possible dereference of a null reference.
+                //         P.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "P").WithLocation(7, 9));
+        }
+
+        [Fact]
+        public void UnassignedStructAutoProperty()
+        {
+            var source =
+@"struct S
+{
+    object? P { get; }
+    S(out object o)
```

Comment:
>S(out object o) [](start = 4, length = 15)

It would be good to test similar scenario in a regular method.

Reply:
Added.